solo/hmac_secret.py

```diff
@@ -86,7 +87,15 @@ def simple_secret(
     client.origin = f"https://{client.host}"
     client.user_id = user_id
     # user = {"id": user_id, "name": "A. User"}
-    credential_id = binascii.a2b_hex(credential_id)
+
+    #check for even number of hex characters (case-insensitive)
+    pattern = re.compile("^([A-Fa-f0-9]{2})+$")
+    if pattern.match(credential_id) is not None:
```

Comment:
Pythonic way is "Easier to ask for forgiveness than permission." (https://docs.python.org/3/glossary.html).

Reply:
okay? actually never done python before this and never was a big fan of throwing exception as it just seems weird and unlogical (and at least in PHP not everything will throw an exception but just break in crazy ways and output a warning, like hex2bin throws false in case something happens so no matter whether you check the input or the output, an if check is going to happen, and if check -> exception -> catch -> error out seems a bit weird compared to just if -> error out)

Reply:
Yeah, but this is Python (whenever you start a new language, idioms change). Possible reading: https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/